r/tools/nixlibs.R

```diff
@@ -435,13 +435,22 @@ turn_off_all_optional_features <- function(env_var_list) {
   # Because these are done as environment variables (as opposed to build flags),
   # setting these to "OFF" overrides any previous setting. We don't need to
   # check the existing value.
+  # Some features turn on other features (e.g. engine -> substrait -> protobuf),
+  # So the list of things to turn off is long. See:
+  # https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L275
   turn_off <- c(
     "ARROW_MIMALLOC" = "OFF",
     "ARROW_JEMALLOC" = "OFF",
     "ARROW_JSON" = "OFF",
     "ARROW_PARQUET" = "OFF", # depends on thrift
     "ARROW_DATASET" = "OFF", # depends on parquet
     "ARROW_S3" = "OFF",
+    "ARROW_GCS" = "OFF",
+    "ARROW_WITH_GOOGLE_CLOUD_CPP" = "OFF",
+    "ARROW_WITH_NLOHMANN_JSON" = "OFF",
+    "ARROW_ENGINE" = "OFF",
+    "ARROW_WITH_SUBSTRAIT" = "OFF",
```

Comment:
ARROW_ENGINE was just renamed

```suggestion
    "ARROW_SUBSTRAIT" = "OFF",
```